code/lighting/lighting_profiles.cpp

```diff
@@ -0,0 +1,166 @@
+#include "globalincs/pstypes.h"
+#include "globalincs/safe_strings.h"
+#include "globalincs/vmallocator.h"
+#include "lighting/lighting.h"
+#include "lighting/lighting_profiles.h"
+#include "parse/parselo.h"
+
+//TODO: maybe a parsehi.cpp would be in order here?
+bool optional_parse_into_float(const SCP_string &fieldname, float* valuetarget)
+{
+	if(optional_string(fieldname.c_str())){
+		stuff_float(valuetarget);
+		return true;
+	}
+	return false;
+}
+
+void lighting_profile::reset()
+{
+	name = "";
+
+    tonemapper = tnm_Uncharted;
+
+	ppc_values.toe_strength = 0.5f;
+    ppc_values.toe_length = 0.5f;
+    ppc_values.shoulder_strength = 0.0f;
+    ppc_values.shoulder_length = 0.5f;
+    ppc_values.shoulder_angle = 0.1f;
+
+	exposure = 4.0f;
+}
+
+TonemapperAlgorithm lighting_profile::name_to_tonemapper(SCP_string &name)
+{
+	SCP_tolower(name);
+	TonemapperAlgorithm r = tnm_Invalid;
+	if(name == "uncharted" || name == "uncharted 2" ){
+		r = tnm_Uncharted;
+	}
+	if(name == "linear"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces approximate"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard jodie"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard extended"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC RGB"){
+		r = tnm_Invalid;
+	}
+	return r;
+}
+
+lighting_profile lighting_profile::default_profile;
+
+void lighting_profile::load_profiles()
+{
+	parse_all();
+}
+
+//The logic for grabbing all the parseable files
+
+void lighting_profile::parse_all()
+{
+	default_profile.reset();
+	if (cf_exists_full("lighting_profiles.tbl", CF_TYPE_TABLES)){
+		mprintf(("TABLES:Starting parse of lighting profiles.tbl"));
+		lighting_profile::parse_file("lighting_profiles.tbl");
+	}
+
+	mprintf(("TBM  =>  Starting parse of lighting profiles ...\n"));
+	parse_modular_table("*-ltp.tbm", lighting_profile::parse_file);
+}
+
+void lighting_profile::parse_default_section()
+{
+	bool keep_going = true;
+	SCP_string buffer;
+	TonemapperAlgorithm tn;
+	while(!optional_string("#END DEFAULT PROFILE") && keep_going){
```

Comment:
Why is this a loop?
Since this is specifically the _default_ profile, it seems to me like it should only be possible to have one entry.